cns/nmagentclient/nmagentclient.go

```diff
@@ -86,3 +96,42 @@ func GetNetworkContainerVersion(
 		networkContainerID, response, err)
 	return response, err
 }
+
+// GetNmAgentSupportedApis :- Retrieves Supported Apis from NMAgent
+func GetNmAgentSupportedApis(
+	getNmAgentSupportedApisURL string) ([]string, string) {
+	var (
+		returnMessage string
+		supportedApis []string
+	)
```

Comment:
In general, there's no need to declare variables far ahead of their usage in Go. `returnMessage` should really be an error that can be inlined in the returns. You should also consider returning explicit zero values where possible. That makes it easy to understand which return paths are happy path. Example:

```go
return nil, fmt.Errorf("failed to retrieve from nmagent: %v", err)
```
is a concise way to wrap an error as well as return the zero value of a slice in non-happy path. In the same manner
```go
return xmlDoc.SupportedApis, nil
```
also makes it clear that no error is being returned to the calling code.

Reply:
Noted.